lib/components/common/PreviewBox.tsx

```diff
@@ -1,19 +1,27 @@
-import React from "react";
 import {Box, Typography, Chip, Button} from "@mui/material";
 import {useDataContext} from "../../contexts/data/DataContext.ts";
 // import PreviewTable from "./PreviewTable";
 import {GlobeIcon, ArrowDropDown, BulletIcon} from "../../icons";
 import {StyledTable} from "./StyledTable";
 import {useUIContext} from "../../contexts/ui/UIContext.ts";
 
+interface PreviewBoxProps {
+    togglePreview: boolean;
+    setTogglePreview: (togglePreview: boolean) => void;
+    onAfterToggle?: () => void;
+}
 
-const PreviewBox = () => {
+const PreviewBox = ({togglePreview, setTogglePreview, onAfterToggle = () => {}}: PreviewBoxProps) => {
```

Comment:
issue (blocking): It seems to me that setTogglePreview and onAfterToggle can be merged into a single prop, for example (onToggle)